### PR TITLE
Convert filter.js to es6 style

### PR DESCRIPTION
**Pull request by killtheliterate**
Because learning.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "asi": false,
-  "node": true,
+  "esnext": true,
```
Comment (LegitTalon) [inline .jshintrc]: What's this do?

File: test/spec/filter.test.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 /* global describe it */
 
-var _ = require('lodash-fp')
-var fixture = require('../fixture/filter-fixture')
-var filter = require('../../src/filter')
-var expect = require('chai').expect
+require('babel/register')
```
Comment (LegitTalon) [inline test/spec/filter.test.js]: I think cause tests/index.js has this you don't need it here? But I could be wrong

Comment (killtheliterate) [inline test/spec/filter.test.js]: right on. I'll take it out.

File: .jshintrc (hunk 1 above)
Comment (killtheliterate) [inline .jshintrc]: makes jshint be okay with es6 syntax.

**Pull request closed without merging**